Run CI on ubicloud self-hosted runners

enescakir opened the pull request:
I run install scripts manually to prepare self-hosted runners. https://github.com/actions/runner-images/blob/main/images/linux/ubuntu2204.pkr.hcl

There are one standard-4 machine at Germany, and one standard-4 machine at Finland.

I attached `ubicloud` and `ubicloud-standard-4` labels to them, so we can use both of them at `runs-on` property.

I added DNS records manually to runners to fix DNS resolving issue on docker containers. Ref: https://github.com/ubicloud/ubicloud/issues/507

Our runners:
<img width="1245" alt="Screenshot 2023-08-25 at 01 25 05" src="https://github.com/ubicloud/ubicloud/assets/993199/4a8683fa-1596-438f-bf4a-182501479a0c">

fdr commented:
Thanks for the detailed commit message, which raises questions (probably @furkansahin shaped)...why are the VMs produced by ubicloud weird enough to require host network exposure to the test program? Is that something strange about us, or the software in question? Exposing the host's network directly is generally seen as some kind of security exposure, not something we'd want to accept lightly and without explanation.

enescakir commented:
> Thanks for the detailed commit message, which raises questions (probably @furkansahin shaped)...why are the VMs produced by ubicloud weird enough to require host network exposure to the test program? Is that something strange about us, or the software in question? Exposing the host's network directly is generally seen as some kind of security exposure, not something we'd want to accept lightly and, in particular, without explanation.

Docker containers can't resolve DNS by default. I fixed it on these runners by adding new DNS servers. Removed `network=host` parameter
https://github.com/ubicloud/ubicloud/issues/507

fdr commented:
> Docker containers can't resolve DNS by default.

Is this true in general (seems odd), or just for us?

enescakir commented:
>  
> 
> > Docker containers can't resolve DNS by default.
> 
> 
> 
> Is this true in general (seems odd), or just for us? It's not so good if docker cannot work quite right by default (I presume this crops up, github action or not)

It not true for local. It depends on networking architecture on cloud.

furkansahin commented:
I like that this runs on our own hosted VM's much quicker than github actions. Drop is around 25%. The Other PRs' CI build takes around 1:10 and here it is 0:52.

enescakir commented:
> I like that this runs on our own hosted VM's much quicker than github actions. Drop is around 25%. The Other PRs' CI build takes around 1:10 and here it is 0:52. 

I created standard-4 machines that have 4vCPUs, GitHub's default runners have 2vCPUs.

furkansahin commented:
Still cheaper isn't it :P

enescakir commented:
Yeah. Even standard-8 and standard-16 are way cheaper
